app/models/test_track/session.rb

```diff
@@ -129,7 +129,7 @@ def create_alias!
   end
 
   def sync_assignments?
-    !visitor.offline? && visitor.unsynced_assignments.present?
+    visitor.loaded? && visitor.unsynced_assignments.present?
```

Comment:
Currently this call will trigger loading the remote assignments, if there were no earlier TT calls in this request. Since plenty of our requests don't have any splits in them, we're unnecessarily hitting TT.

I'm not sure I like exposing a `loaded?` method, it leaks too much of the internals of visitor into session. What about a method like `has_unsynced_assignments?` that will not trigger loading the remote assignments?